R/read_lookup_sc_client.R

```diff
@@ -3,17 +3,15 @@
 #' @description This will read and process the
 #' social care client lookup
 #'
+#' @param sc_dvprod_connection The connection to the SC platform.
 #' @param year The year to process
 #'
 #' @return the final data as a [tibble][tibble::tibble-package].
 #' @export
 #' @family process extracts
-read_lookup_sc_client <- function(year) {
-  # set-up conection to platform
-  db_connection <- phs_db_connection(dsn = "DVPROD")
-
+read_lookup_sc_client <- function(sc_dvprod_connection = phs_db_connection(dsn = "DVPROD"), year = convert_fyyear_to_year(year)) {
```

Comment:
This doesn't make sense to have the default value for a parameter also contain that parameter.

We can, however, just take a year in fyyear format (and check this is the case), then convert it to the calendar year format we need before using it in the `dbplyr` code

```suggestion
read_lookup_sc_client <- function(sc_dvprod_connection = phs_db_connection(dsn = "DVPROD"), fyyear) {

check_year_format(fyyear)
year <- convert_fyyear_to_year(fyyear)
```